tests/e2e/workloads/amq/test_rgw_kafka_notifications.py

```diff
@@ -0,0 +1,185 @@
+        """
+        Test to verify rgw kafka notifications
+
+        """
+        # Get sc
+        sc = default_storage_class(interface_type=constants.CEPHBLOCKPOOL)
+
+        # Deploy amq cluster
+        self.amq.setup_amq_cluster(sc.name)
+
+        # Create topic
+        self.kafka_topic = self.amq.create_kafka_topic()
+
+        # Create kafkadrop pod
+        try:
+            kafkadrop = list(
+                templating.load_yaml(constants.KAFKADROP_YAML, multi_document=True)
+            )
+            self.kafkadrop_pod = OCS(**kafkadrop[0])
+            self.kafkadrop_svc = OCS(**kafkadrop[1])
+            self.kafkadrop_route = OCS(**kafkadrop[2])
+            self.kafkadrop_pod.create()
+            self.kafkadrop_svc.create()
+            self.kafkadrop_route.create()
+        except (CommandFailed, CalledProcessError) as cf:
+            log.error("Failed during creation of kafkadrop which kafka UI")
+            raise cf
+
+        # Validate kafkadrop pod running
+        ocp_obj = OCP(kind=constants.POD, namespace=constants.AMQ_NAMESPACE)
+        ocp_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector="app=kafdrop",
+            timeout=120,
+            sleep=5,
+        )
+
+        # Get the kafkadrop route
+        kafkadrop_host = self.kafkadrop_route.get().get("spec").get("host")
+
+        # # Get the kafkaendpoint
+        kafka_endpoint = f"my-cluster-kafka-bootstrap.{constants.AMQ_NAMESPACE}.svc.cluster.local:9092"
+
+        # Create bucket
+        bucketname = bucket_factory(amount=1, interface=interface)[0].name
+
+        # Get RGW credentials
+        rgw_obj = RGW()
+        rgw_endpoint, access_key, secret_key = rgw_obj.get_credentials()
+
+        # Clone notify repo
+        notify_path = clone_notify()
+
+        # Initialise to put objects
+        data = "A random string data to write on created rgw bucket"
+        obc_obj = OBC(bucketname)
+        s3_resource = boto3.resource(
+            "s3",
+            verify=retrieve_verification_mode(),
+            endpoint_url=rgw_endpoint,
+            aws_access_key_id=obc_obj.access_key_id,
+            aws_secret_access_key=obc_obj.access_key,
+        )
+        s3_client = s3_resource.meta.client
+
+        # Initialize notify command to run
+        notify_cmd = (
+            f"python {notify_path} -e {rgw_endpoint} -a {obc_obj.access_key_id} "
+            f"-s {obc_obj.access_key} -b {bucketname} -ke {kafka_endpoint} -t {self.kafka_topic.name}"
+        )
+        log.info(f"Running cmd {notify_cmd}")
+
+        # Put objects to bucket
+        assert s3_client.put_object(
+            Bucket=bucketname, Key="key-1", Body=data
+        ), "Failed: Put object: key-1"
+        run_cmd(notify_cmd)
+
+        # Validate rgw logs notification are sent
+        # No errors are seen
+        pattern = "ERROR: failed to create push endpoint"
+        rgw_pod_name = get_pod_name_by_pattern(
```

Comment:
You may use this get_rgw_pods()  https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/pod.py#L612